x/dclupgrade/keeper/msg_server_propose_upgrade.go

```diff
@@ -45,6 +33,20 @@ func (k msgServer) ProposeUpgrade(goCtx context.Context, msg *types.MsgProposeUp
 		return nil, err
 	}
 
+	// check if proposed upgrade with the same name already exists and propose plan height
+	existProposedUpgrade, isFound := k.GetProposedUpgrade(ctx, msg.Plan.Name)
```

Comment:
Better name would be `existingProposedUpgrade`

Reply:
Done! ✅